### ISSUE TITLE
A ColorSpaceTransform from/to the same ColorSpace should always be a no-op

### ISSUE THREAD

**Issue by doug-walker** [AcademySoftwareFoundation/OpenColorIO#602]
Somewhat surprisingly, the function AreColorSpacesInSameEqualityGroup( xyz, xyz ) in ColorSpaceTransform.cpp returns false if there is no equalitygroup defined for space xyz.

Because BuildColorSpaceOps uses this function to decide whether to treat the conversion as a no-op, the fact that a color space is not considered equal to itself produces some odd results.  For example:

```
  - !<ColorSpace>
    name: A
    equalitygroup: A

  - !<ColorSpace>
    name: AA
    equalitygroup: A

  - !<ColorSpace>
    name: B
    equalitygroup: 

```
```
!<ColorSpaceTransform> {src: A, dst: AA}  is always a no-op whereas
!<ColorSpaceTransform> {src: B, dst: B}  may not be a no-op (depending on how its components are optimized at the opVec level)

```
The argument has been made on Slack that if the round trip B-->reference, reference-->B is lossy then ColorSpaceTransform {B, B} should also be lossy.  However, the counter-argument is that OCIO's equalitygroup feature has already established the notion that transforms between "equivalent" color spaces should be evaluated as no-ops.  

For OCIO v1, a workaround if one wants {src: B, dst: B} to be a no-op is to create an "equalitygroup of one" for each color space (containing itself).  

I'm creating this issue to have a place to discuss what the behavior should be for OCIO v2.  The proposal is that {src: B, dst: B} should always be considered a no-op at the BuildColorSpaceOps level.

**Comment by doug-walker** [comment]
Last call for any comments/discussion on this issue.  As we work on implementing issue #1129 (Aliases), this issue becomes more relevant.  Our opinion is that a transform from a color space to one of its aliases should be a no-op, rather than a conversion from a color space to its reference and then back to the same color space.  We plan on fixing this for v2 unless there are any objections.

**Comment by doug-walker** [comment]
Last call for any comments/discussion on this issue.  As we work on implementing issue #1129 (Aliases), this issue becomes more relevant.  Our opinion is that a transform from a color space to one of its aliases should be a no-op, rather than a conversion from a color space to its reference and then back to the same color space.  We plan on fixing this for v2 unless there are any objections.

**Comment by doug-walker** [comment]
Closing as this was implemented as part of PR #1241.
